app/src/main/java/org/dhis2/utils/filters/FilterManager.java

```diff
@@ -218,14 +218,14 @@ public void addEnrollmentStatus(boolean remove, EnrollmentStatus enrollmentStatu
     public void addPeriod(List<DatePeriod> datePeriod) {
         this.periodFilters = datePeriod;
 
-        periodFiltersApplied.set(datePeriod != null ? 1 : 0);
+        periodFiltersApplied.set(datePeriod != null ? datePeriod.size() : 0);
```

Comment:
These changes results in the unit test  to fail.
You can only have one active period at a time. So the related issue should not be fixed this way.